com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -266,14 +269,72 @@ internal string GetSceneNameFromPath(string scenePath)
         }
 
         /// <summary>
-        /// Generates the scenes in build list
+        /// Generates the hash values and associated tables
+        /// for the scenes in build list
         /// </summary>
         internal void GenerateScenesInBuild()
         {
-            ScenesInBuild.Clear();
+            HashToBuildIndex.Clear();
+            BuildIndexToHash.Clear();
             for (int i = 0; i < SceneManager.sceneCountInBuildSettings; i++)
             {
-                ScenesInBuild.Add(GetSceneNameFromPath(SceneUtility.GetScenePathByBuildIndex(i)));
+                var scenePath = SceneUtility.GetScenePathByBuildIndex(i);
+                var hash = XXHash.Hash32(scenePath);
+                var buildIndex = SceneUtility.GetBuildIndexByScenePath(scenePath);
+                HashToBuildIndex.Add(hash, buildIndex);
```

Comment:
There's coupling between the two lookup structures. They should always be consistent with each other. To guarantee that this holds, one could have a `BidirectionalDictionary`. I'm too sure it's a good idea, its extra complexity, but it would guarantee consistency between `HashToBuildIndex` and `BuildIndexToHash`.

Alternatively, we could add for non-ship builds a check done regularly that verify the two mapping are consistent.

Reply:
Well, I understand this desire, but there is only one time when this is generated...and it is done all inside that one method.  For V1.0.0. once generated, the intention is that it will not change for the duration of that session.  There is no official BidirectionalDictionary, but it is basically the same thing with added checks if something changes.  These tables, for V1.0,0, should not change.